### PR TITLE
upgrade to next15

### PR DESCRIPTION
**Pull request by adriangohjw**
## Problem

Hoping to leverage the performance improvement stated in https://nextjs.org/blog/next-15#faster-static-generation-for-the-app-router to speed up SSG which is the remaining main bottleneck of slow build

partially closes https://linear.app/ogp/issue/ISOM-1661/optimise-site-build-times

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [ ] No - this PR is backwards compatible

**Improvements**:

- improved codebuild `next build` by 3 minutes (540s -> 360s) when tested on `moh-corp` PROD

note:
- only upgrading for `tooling/template` because next 15 is not supported by storybook used in studio at the moment. Thus we have to tell the lint to exclude it

## Tests

**New dependencies**:

- `dependency` : `next` to v15

### CODE REVIEW

**Comment by adriangohjw** [comment]
need to put on HOLD because storybook is not compatible with next15 yet https://github.com/storybookjs/storybook/issues/29421

File: package.json
Comment (harishv7) [inline package.json:21]: can i clarify what -i next does?

File: tooling/template/app/[[...permalink]]/page.tsx
Comment (harishv7) [inline tooling/template/app/[[...permalink]]/page.tsx:15]: do we need this statement across the files? could you summarise what this implies?

Comment (harishv7) [reply tooling/template/app/[[...permalink]]/page.tsx:15]: I would think that having the following in the next.config will auto-force all to be static

```
const nextConfig = {
  output: "export",
```

File: package.json
Comment (adriangohjw) [reply package.json:21]: it tells `sherif` to ignore the different `next` versions and not raise it as error

this is needed because `postinstall` runs `lint:ws` and block `npm i` from proceeding

File: tooling/template/app/[[...permalink]]/page.tsx
Comment (adriangohjw) [reply tooling/template/app/[[...permalink]]/page.tsx:15]: @harishv7 yeah i would think so too but somehow not including this will cause an issue during `next build`

**Review by harishv7 — APPROVED** [review]
@adriangohjw  approving, please test on our repos before merging to ensure builds succeed

**Pull request merged**
